### DISCUSSION TITLE
Key Sign Appcelerator

### DISCUSSION THREAD

**Discussion by matiassepu** [appcelerator/titanium_mobile#12629] [Q&A] [answered]
Hi, does anyone know what kind of signature Appcelerator uses? I am migrating an app from Appcelerator to java, I sign the application with the same signature used in the Appcelerator version but when installing the apk on a device with ADB it throws me a signature error

**Comment by jquick-axway** [comment, marked as answer]
Titanium only uses its own keystore for "debug" builds which can't be uploaded to Google Play. Titanium's debug keystore info is below. Just note that this is **never** used for release builds.
- Keystore File Location: `~/Library/Application Support/Titanium/mobilesdk/osx/<version>/android/dev_keystore`
- Keystore Password: `tirocks`
- Alias Name: `tidev`
- Alias Password: `tirocks`

In order to do release/production builds with Titanium, you have to supply your own keystore file, password, alias name, and alias password. Titanium does not automatically create this for you. You must have done this at some point. So, whatever you set those to you need to assign them to your native Android app project's "build.gradle" file.
```groovy
android {
	signingConfigs {
		config {
			storeFile file(yourKeystoreFilePath)
			storePassword yourKeystorePassword
			keyAlias yourKeystoreAliasName
			keyPassword yourKeystoreAliasPassword
		}
	}
}
```

When you do a release build via Android Studio, it'll ask you for the above info.

**Comment by matiassepu** [comment]
I have signed my app with my own key, not the debug one, and with this same key I sign the app in Android Studio (my own key). What I discovered is that in Google Play the app was signed with another key, that's why it gave the error. I've already corrected it, thanks anyway.
